### PR TITLE
Change BIER BitString field to display as hex.

### PR DESCRIPTION
**Pull request by pvinci**

### CODE REVIEW

File: scapy/contrib/bier.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 from scapy.packet import Packet, bind_layers
 from scapy.fields import BitEnumField, BitField, BitFieldLenField, ByteField, \
-    ShortField, StrLenField
+    ShortField, StrLenField, XStrLenField
```
Comment (guedou) [inline scapy/contrib/bier.py]: StrLenField must be removed (there is a flake8 error message https://travis-ci.com/secdev/scapy/jobs/268379551#L657)

File: scapy/contrib/bier.py [hunk 2]
```diff
@@ -58,9 +58,9 @@ class BIER(Packet):
                    BitField("DSCP", 0, 6),
                    BitEnumField("Proto", 2, 6, BIERnhcls),
                    ShortField("BFRID", 0),
-                   StrLenField("BitString",
-                               "",
-                               length_from=lambda x:(8 << x.length))]
+                   XStrLenField("BitString",
+                                "",
```
Comment (guedou) [inline scapy/contrib/bier.py]: Could you explain why you are changing this field?
Also, changes lines 62 & 63 are not necessary.

File: scapy/contrib/cdp.py [hunk 3]
```diff
@@ -357,7 +357,7 @@ def _check_len(self, pkt):
         This padding is only used for checksum computation.  The original
         packet should not be altered."""
         if len(pkt) % 2:
-            last_chr = pkt[-1]
+            last_chr = pkt[len(pkt) - 1:]
```
Comment (guedou) [inline scapy/contrib/cdp.py]: How is this related to the change of the field?

Comment (pvinci) [reply scapy/contrib/cdp.py]: This is a separate change.  I need to work through the details of the pull request workflow.  I will fix.

**Comment by guedou** [comment]
Closing this PR: it does not seem necessary, and the author did not provide answers.
